src/main/java/org/opentripplanner/apis/transmodel/support/OneOfInputValidator.java

```diff
@@ -0,0 +1,60 @@
+package org.opentripplanner.apis.transmodel.support;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Validate @oneOf directive, this validation is NOT done by the Java GraphQL library at the
+ * moment(v22.1). Remove this when enforced by the library. The {@code @oneOf} is an experimental
+ * feature in this version of the library.
+ * <p>
+ * See {@link graphql.Directives#OneOfDirective}
+ */
+public class OneOfInputValidator {
```

Comment:
The changelog for graphql-java 22.2 mentions some improvements for `@oneOf`: https://github.com/graphql-java/graphql-java/releases/tag/v22.2

We have been using it in the GTFS API for a while so I'm pretty sure it has been around in some form.

Reply:
There is support for the @oneOf - the directive is part of the Java lib, but the lib does not enforce it. I can guess a few reasons why, but I have not checked. I did test without this, an it is not enforced. The correct way to implement this is to plug it into the framework, but I am hoping I do not need to do that and that it will be enforced by the lig, when @oneOf become a ofical feature.

Reply:
I investigated this and it seems it's a difference between using the schema-first workflow (GTFS) and building the schema in code (Transmodel), because in GTFS it _does_ actually enforce it.

### GTFS

![Screenshot from 2024-09-26 22-25-54](https://github.com/user-attachments/assets/bd5da8fc-123f-4434-b035-60b6b2389492)

Here _the library_ correctly rejects the input.

### Transmodel

![Screenshot from 2024-09-26 22-27-17](https://github.com/user-attachments/assets/4e005aa0-641d-47e9-a771-82c976a46695)